packages/utils/telemetry-utils/src/logger.ts

```diff
@@ -218,6 +197,55 @@ export abstract class TelemetryLogger implements ITelemetryLogger {
     }
 }
 
+/**
+ * TaggedLoggerAdapter class can add tag handling to your logger.
+ */
+ export class TaggedLoggerAdapter implements ITelemetryBaseLogger {
+    public constructor(
+        private readonly logger: ITelemetryBaseLogger) {
+    }
+
+    public send(eventWithTagsMaybe: ITelemetryBaseEvent) {
+        const newEvent: ITelemetryBaseEvent = {
+            category: eventWithTagsMaybe.category,
+            eventName: eventWithTagsMaybe.eventName,
+        };
+        for (const key of Object.keys(getValidTelemetryProps(eventWithTagsMaybe, new Set()))) {
+            if (newEvent[key] !== undefined) {
+                // Don't overwrite existing properties on the event
+                continue;
+            }
```

Comment:
nit: I don't think this is necessary - newEvent starts empty except for category and eventName, and it's fine for them to get overwritten with the same value IMO.